src/collection.js

```diff
@@ -44,143 +109,422 @@ export default class Collection {
-    });
+    }).catch(this._handleError("get"));
   }
 
-  delete(id) {
+  /**
+   * Deletes a record from the local database.
+   *
+   * @param  {String} id
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  delete(id, options={virtual: true}) {
     return this.open().then(() => {
       // Ensure the record actually exists.
-      return this.get(id).then(result => {
+      return this.get(id, {includeDeleted: true}).then(res => {
+        if (options.virtual) {
+          if (res.data._status === "deleted") {
+            // Record is already deleted
+            return Promise.resolve({
+              data: { id: id },
+              permissions: {}
+            });
+          } else {
+            return this.update(Object.assign({}, res.data, {
+              _status: "deleted"
+            }));
+          }
+        }
         return new Promise((resolve, reject) => {
           const {transaction, store} = this.prepare("readwrite");
           store.delete(id);
-          transaction.oncomplete = function(event) {
+          transaction.onerror = event => reject(new Error(event.target.error));
+          transaction.oncomplete = event => {
             resolve({
-              data: { id: id, deleted: true },
+              data: { id: id },
               permissions: {}
             });
           };
-          transaction.onerror = function(event) {
-            reject(new Error(event.target.error));
-          };
         });
       });
-    });
+    }).catch(this._handleError("delete"));
   }
 
-  list() {
+  /**
+   * Lists records from the local database.
+   *
+   * @param  {Object} params
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  list(params={}, options={includeDeleted: false}) {
     return this.open().then(() => {
       return new Promise((resolve, reject) => {
-        var results = [];
+        const results = [];
         const {transaction, store} = this.prepare();
-        var request = store.openCursor();
+        const request = store.openCursor();
         request.onsuccess = function(event) {
           var cursor = event.target.result;
           if (cursor) {
-            results.push(cursor.value);
+            if (options.includeDeleted || cursor.value._status !== "deleted") {
+              results.push(cursor.value);
+            }
             cursor.continue();
           }
         };
-        transaction.oncomplete = function(event) {
+        transaction.onerror = event => reject(new Error(event.target.error));
+        transaction.oncomplete = event => {
           resolve({
             data: results,
             permissions: {}
           });
         };
-        transaction.onerror = function(event) {
-          reject(new Error(event.target.error));
+      });
+    }).catch(this._handleError("list"));
+  }
+
+  /**
+   * Import a single change into the local database.
+   *
+   * @param  {Object} change
+   * @return {Promise}
+   */
+  importChange(change) {
```

Comment:
This could be private API ? since it is only used by ``ìmportChanges()``
``_importChange()`` ?